src/mod/endpoints/mod_sofia/sofia_reg.c

```diff
@@ -3408,15 +3412,15 @@ switch_status_t sofia_reg_add_gateway(sofia_profile_t *profile, const char *key,
 	switch_mutex_unlock(profile->gw_mutex);
 
 	switch_mutex_lock(mod_sofia_globals.hash_mutex);
-
 	if ((gp = switch_core_hash_find(mod_sofia_globals.gateway_hash, key))) {
-		if (gp->deleted) {
+		switch_mutex_lock(profile->gw_deleting_mutex);
```

Comment:
Move this outside in the if condition otherwise a potential crash due to gp might be a garbage pointer

Reply:
done, thank you for the observation